consensus/dummy/dynamic_fees.go

```diff
@@ -83,65 +69,55 @@ func CalcBaseFee(config *params.ChainConfig, parent *types.Header, timestamp uin
-	// gas in.
-	if roll < params.RollupWindow {
-		var blockGasCost, parentExtraStateGasUsed uint64
-		switch {
-		case isApricotPhase5:
-			// [blockGasCost] has been removed in AP5, so it is left as 0.
-
-			// At the start of a new network, the parent
-			// may not have a populated [ExtDataGasUsed].
-			if parent.ExtDataGasUsed != nil {
-				parentExtraStateGasUsed = parent.ExtDataGasUsed.Uint64()
-			}
-		case isApricotPhase4:
-			// The [blockGasCost] is paid by the effective tips in the block using
-			// the block's value of [baseFee].
-			blockGasCost = calcBlockGasCost(
-				ApricotPhase4TargetBlockRate,
-				ApricotPhase4MinBlockGasCost,
-				ApricotPhase4MaxBlockGasCost,
-				ApricotPhase4BlockGasCostStep,
-				parent.BlockGasCost,
-				parent.Time, timestamp,
-			).Uint64()
+	// Add in parent's consumed gas
+	var blockGasCost, parentExtraStateGasUsed uint64
+	switch {
+	case isApricotPhase5:
+		// [blockGasCost] has been removed in AP5, so it is left as 0.
 
-			// On the boundary of AP3 and AP4 or at the start of a new network, the parent
-			// may not have a populated [ExtDataGasUsed].
-			if parent.ExtDataGasUsed != nil {
-				parentExtraStateGasUsed = parent.ExtDataGasUsed.Uint64()
-			}
-		default:
-			blockGasCost = ApricotPhase3BlockGasFee
+		// At the start of a new network, the parent
+		// may not have a populated [ExtDataGasUsed].
+		if parent.ExtDataGasUsed != nil {
+			parentExtraStateGasUsed = parent.ExtDataGasUsed.Uint64()
 		}
-
-		// Compute the new state of the gas rolling window.
-		addedGas, overflow := math.SafeAdd(parent.GasUsed, parentExtraStateGasUsed)
-		if overflow {
-			addedGas = math.MaxUint64
+	case isApricotPhase4:
+		// The [blockGasCost] is paid by the effective tips in the block using
+		// the block's value of [baseFee].
+		blockGasCost = calcBlockGasCost(
+			ApricotPhase4TargetBlockRate,
+			ApricotPhase4MinBlockGasCost,
+			ApricotPhase4MaxBlockGasCost,
+			ApricotPhase4BlockGasCostStep,
+			parent.BlockGasCost,
+			parent.Time, timestamp,
+		).Uint64()
+
+		// On the boundary of AP3 and AP4 or at the start of a new network, the parent
+		// may not have a populated [ExtDataGasUsed].
+		if parent.ExtDataGasUsed != nil {
+			parentExtraStateGasUsed = parent.ExtDataGasUsed.Uint64()
 		}
+	default:
+		blockGasCost = ApricotPhase3BlockGasFee
+	}
 
-		// Only add the [blockGasCost] to the gas used if it isn't AP5
-		if !isApricotPhase5 {
-			addedGas, overflow = math.SafeAdd(addedGas, blockGasCost)
-			if overflow {
-				addedGas = math.MaxUint64
-			}
-		}
+	// Compute the new state of the gas rolling window.
+	dynamicFeeWindow.Add(parent.GasUsed, parentExtraStateGasUsed, blockGasCost)
 
-		slot := params.RollupWindow - 1 - roll
-		start := slot * wrappers.LongLen
-		updateLongWindow(newRollupWindow, start, addedGas)
+	if timestamp < parent.Time {
+		return nil, nil, fmt.Errorf("cannot calculate base fee for timestamp (%d) prior to parent timestamp (%d)", timestamp, parent.Time)
```

Comment:
nit why do we use parentheses for the timestamp numbers?

Reply:
🤷 that's just how it was. I removed the parens.